test/tbb/test_overwrite_node.cpp

```diff
@@ -230,3 +230,33 @@ TEST_CASE("try_release"){
 
     CHECK_MESSAGE ((on.try_release()== true), "try_release should return true");
 }
+
+template< typename Type >
+struct fake_receiver : public tbb::flow::receiver<Type>, utils::NoAssign {
+    tbb::flow::graph& my_graph;
+
+    fake_receiver(tbb::flow::graph& g) : my_graph(g) {}
+
+    tbb::detail::d1::graph_task* try_put_task(const Type&) override {
+        return nullptr;
+    }
```

Comment:
If I remember correctly, returning `nullptr` is the very reason to turn over the edge state from push to pull, which implies spawning of a `register_predecessor_task`. (BTW, since `overwrite_node` here is a predecessor and `fake_receiver` is its successor, why not it is called a `register_successor_task`?)
However, I cannot understand why that spawned task becomes cancelled? Could you please remind to me?
Writing a detailed comment for this test to clarify and explain the details to its future readers would even be better.

Reply:
Updated with using `join_node` instead of custom receiver.

To spawn `register_predecessor_task` of `overwrite_node` internal buffer should be valid (calling `overwrite_node::try_put`) and successor should return `nullptr` during call `try_put_task`.

Since group context is canceled by calling `graph::cancel()`, all spawned tasks will be cancelled during `graph::wait_for_all`.  `graph::wait_for_all` may be directly called like in test or as part of `graph` destructor.

Reply:
> Since group context is canceled by calling graph::cancel(), all spawned tasks will be cancelled 

Is it guaranteed? It seems to me that they can also be executed once spawned, even before graph.cancel() or even graph.wait_for_all(), cannot they?

Reply:
Race is possible. Updating review by moving `graph.cancel()` before task spawn. Task will be cancelled by worker or external thread.